### PR TITLE
Update to .NET 10 build images

### PR DESCRIPTION
**Pull request by sbomer**
This updates the libc requirement to:
- glibc 2.27
- musl libc 1.2.3

These images have LLVM 19.

Contributes to https://github.com/dotnet/runtime/issues/109939

### CODE REVIEW

File: eng/pipelines/common/templates/pipeline-with-resources.yml [hunk 1]
```diff
@@ -27,44 +27,44 @@ extends:
...
 
       linux_arm64:
-        image: mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net9.0-cross-arm64
+        image: mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-cross-arm64
         env:
           ROOTFS_DIR: /crossrootfs/arm64
 
       linux_musl_x64:
-        image: mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net9.0-cross-amd64-alpine
+        image: mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-cross-amd64-alpine
```
Comment (richlander) [inline eng/pipelines/common/templates/pipeline-with-resources.yml:35]: I thought we were going to move to `-musl`. Wouldn't it make sense to block this PR on that change?

**Comment by jkoritzinsky** [comment]
Looks like there's some issues with the ARM32 images with zlib-ng, in particular around intrinsics.

Other than that, this looks good to me.

File: eng/pipelines/common/templates/pipeline-with-resources.yml [hunk 2]
```diff
@@ -109,17 +109,17 @@ extends:
...
 
       wasi_wasm:
-        image: mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net9.0-webassembly-amd64
+        image: mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-webassembly-amd64
         env:
           ROOTFS_DIR: /crossrootfs/x64
 
       freebsd_x64:
-        image: mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net9.0-cross-freebsd-13
+        image: mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-cross-freebsd-13-amd64
```
Comment (am11) [inline eng/pipelines/common/templates/pipeline-with-resources.yml:122]: We have removed v13 images (i.e. existing images will stay in ACR, no new image will be produced)
```suggestion
        image: mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-cross-freebsd-14-amd64
```

File: docs/workflow/using-docker.md [hunk 3]
```diff
@@ -26,26 +26,26 @@ The main Docker images are the most commonly used ones, and the ones you will pr
...
-| Azure Linux (x64) | Ubuntu 18.04               | S390x         | `mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net9.0-cross-s390x`         | `/crossrootfs/s390x`   |
-| Azure Linux (x64) | Ubuntu 16.04 (Wasm)        | x64           | `mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net9.0-webassembly-amd64`   | `/crossrootfs/x64`     |
+| Azure Linux (x64) | Android Bionic             | x64           | `mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-cross-android-amd64` |        *N/A*           |
+| Azure Linux (x64) | Android Bionic (w/OpenSSL) | x64           | `mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-android-openssl`     |        *N/A*           |
+| Azure Linux (x64) | Android Bionic (w/Docker)  | x64           | `mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-android-docker`      |        *N/A*           |
+| Azure Linux (x64) | FreeBSD 13                 | x64           | `mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-cross-freebsd-13`    | `/crossrootfs/x64`     |
```
Comment (am11) [inline docs/workflow/using-docker.md:44]: likewise
```suggestion
| Azure Linux (x64) | FreeBSD 14                 | x64           | `mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-cross-freebsd-14`    | `/crossrootfs/x64`     |
```

**Comment by am11** [comment]
alpine arm32 build fixes: https://github.com/dotnet/runtime/compare/main...am11:runtime:feature/native/build-fixes

**Review by akoeplinger — APPROVED** [review]

**Comment by akoeplinger** [comment]
/ba-g test timeout is due to https://github.com/dotnet/runtime/issues/110285 which is fixed in main and closed so Build Analysis didn't link it.

**Comment by ericstj** [comment]
@akoeplinger signed off and flipped to green, @sbomer is this ready to merge?

**Comment by akoeplinger** [comment]
According to Outlook he's OOF until 12/19

**Pull request merged**